marshmallow_recipe/bake.py

```diff
@@ -95,21 +96,28 @@ def get_field_for(
             raise ValueError(f"Unsupported {type=}")
         required = False
         type = next(type_arg for type_arg in type_args if type_arg is not types.NoneType)  # noqa
+    elif "default" in metadata and metadata["default"] is not dataclasses.MISSING:
+        required = False
     else:
         required = True
+    if type_from_union is dataclasses.MISSING:
+        allow_none = False
+    else:
```

Comment:
I am slightly unhappy with the change: allow_none seems to be set to True for non optional fields.

Reply:
There is a dedicated test about the case
https://github.com/anna-money/marshmallow-recipe/pull/76/files#diff-f1dbe79adfb6e39e76ee4519368bc4b5eaa0e4831f00165e5ac28fede4825bffR111

Reply:
Sorry, my fault